Modules/Challenges/test/Challenges.API.Tests.Integration/StepDefinitions/ChallengesApiStepDefinitions.cs

```diff
@@ -0,0 +1,157 @@
+using Challenges.API.Tests.Integration.API;
+using Challenges.API.Tests.Integration.Extensions;
+using Challenges.API.Tests.Integration.Models;
+using Microsoft.Extensions.Options;
+using TechTalk.SpecFlow.Assist;
+using static Challenges.API.Tests.Integration.Configuration.Settings;
+
+namespace Challenges.API.Tests.Integration.StepDefinitions;
+
+[Binding]
+public class ChallengesApiStepDefinitions
+{
+    private readonly ChallengesApi _challengeApi;
+    private string _challengeId;
+    private HttpResponse<ChallengeResponse> _challengeResponse;
+    private readonly RequestConfiguration _requestConfiguration;
+
+    public ChallengesApiStepDefinitions(IOptions<HttpConfiguration> httpConfiguration, ChallengesApi challengeApi)
+    {
+        _challengeApi = challengeApi;
+        _challengeId = string.Empty;
+        _challengeResponse = new HttpResponse<ChallengeResponse>();
+        _requestConfiguration = new RequestConfiguration
+        {
+            AuthenticationParameters = new AuthenticationParameters
+            {
+                GrantType = "password",
+                ClientId = httpConfiguration.Value.ClientCredentials.ClientId,
+                ClientSecret = httpConfiguration.Value.ClientCredentials.ClientSecret,
+                UserName = "USRa",
+                Password = "a"
+            }
+        };
+    }
+
+    [Given(@"the user is authenticated")]
+    public void GivenTheUserIsAuthenticated()
+    {
+        _requestConfiguration.IsAuthenticated = true;
+    }
+
+    [Given(@"the user is unauthenticated")]
+    public void GivenTheUserIsUnauthenticated()
+    {
+        _requestConfiguration.IsAuthenticated = false;
+    }
+
+    [Given(@"the Accept header is '([^']*)'")]
+    public void GivenTheAcceptHeaderIs(string acceptHeader)
+    {
+        _requestConfiguration.AcceptHeader = acceptHeader;
+    }
+
+    [Given(@"a Challenge c")]
+    public async Task GivenAChallengeC()
+    {
+        var challengeResponse = await _challengeApi.CreateChallenge(_requestConfiguration);
+        challengeResponse.IsSuccessStatusCode.Should().BeTrue();
+
+        _challengeId = challengeResponse.Data!.Result!.Id;
+        _challengeId.Should().NotBeNullOrEmpty(because: "Required value for 'Id' is missing.");
```

Comment:
I would not assert this here. That's not the responsibility of this step.

Reply:
Alright. I think we even discussed this some days ago, and you already convinced me back then. Sorry for repeating this. :)